lineman/app/components/change_password_form/change_password_form.coffee

```diff
@@ -0,0 +1,17 @@
+angular.module('loomioApp').factory 'ChangePasswordForm', ->
+  templateUrl: 'generated/components/change_password_form/change_password_form.html'
+  controller: ($scope, $rootScope, CurrentUser, Records, FlashService) ->
+    $scope.user = CurrentUser.clone()
+
+    $scope.submit = ->
+      $scope.isDisabled = true
+      Records.users.changePassword($scope.user).then ->
+        $scope.isDisabled = false
+        FlashService.success('profile_page.messages.password_changed')
+        $scope.$close()
+      , (response) ->
+        $scope.isDisabled = false
+        if response.status == 422
```

Comment:
Yeah I've been thinking there's a lot of duplication between #save and the
records interface stuff; I'd be cool with moving all the network calls to
the record interface.

On Wed, Jul 8, 2015 at 4:08 AM, Robert Guthrie <notifications@github.com>
wrote:

> In lineman/app/components/change_password_form/change_password_form.coffee
> <https://github.com/loomio/loomio/pull/2282#discussion_r34136648>:
>
> > @@ -0,0 +1,17 @@
> > +angular.module('loomioApp').factory 'ChangePasswordForm', ->
> > +  templateUrl: 'generated/components/change_password_form/change_password_form.html'
> > +  controller: ($scope, $rootScope, CurrentUser, Records, FlashService) ->
> > +    $scope.user = CurrentUser.clone()
> > +
> > +    $scope.submit = ->
> > +      $scope.isDisabled = true
> > +      Records.users.changePassword($scope.user).then ->
> > +        $scope.isDisabled = false
> > +        FlashService.success('profile_page.messages.password_changed')
> > +        $scope.$close()
> > +      , (response) ->
> > +        $scope.isDisabled = false
> > +        if response.status == 422
>
> Yea I think it's pretty simple. I like the submit, isDisabled, close
> simplicity.
>
> I think setErrors should be done within the records_interface function
> with chained promises. (like save() on Model)
>
> With all the network calls moving to the record_interface I'm thinking we
> should consider moving base_model#save to the records_interface then we
> wouldn't need restful_client in the model at all. Then we separate the
> network stuff from the local stuff (find, where) completely?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/loomio/loomio/pull/2282/files#r34136648>.
>